firmware/src/widgets/clockWidget.cpp

```diff
@@ -114,33 +109,58 @@ void ClockWidget::buttonPressed(uint8_t buttonId, ButtonState state) {
         changeMode();
 }
 
-void ClockWidget::displayDigit(int displayIndex, const String& digit, int font, int fontSize, uint32_t color, bool shadowing) {
+DigitOffset ClockWidget::getOffsetForDigit(const String &digit) {
+    if (digit.length() == 0) {
+        return {0, 0};
+    }
+    char c = digit.charAt(0);
+    if (c >= 48 && c <= 57) {
+        return m_digitOffsets[c-48];
+    } else {
+        return {0, 0};
+    }
```

Comment:
This should probably come from the font.

Reply:
I don't know how, but I'm open to suggestions.